cmd/pmm-configurator/main.go

```diff
@@ -49,7 +54,29 @@ func main() {
 	log.WithFields(log.Fields{
 		"address": c.ListenAddress,
 	}).Info("PMM Configurator is started")
-	log.Fatal(http.ListenAndServe(c.ListenAddress, router))
+
+	hs := &http.Server{Addr: c.ListenAddress, Handler: router}
+	gracefulStop(hs)
+	if err := hs.ListenAndServe(); err != http.ErrServerClosed {
+		log.Fatal(err)
+	}
+}
+
+func gracefulStop(hs *http.Server) {
+	var gracefulStop = make(chan os.Signal)
+	signal.Notify(gracefulStop, syscall.SIGINT)
+	signal.Notify(gracefulStop, syscall.SIGTERM)
+	go func() {
+		<-gracefulStop
+
+		if err := hs.Shutdown(context.Background()); err != nil {
+			log.Errorf("%v", err)
+		} else {
+			log.Info("Server stopped")
+		}
+		time.Sleep(2 * time.Second)
```

Comment:
?

Reply:
Advanced Sleep Programming 🙈
sleep is needed to give time for HTTP Server to stop in another thread.
unfortunately, without clean shutdown - coverage report cannot be generated.

Reply:
Sorry, but well, I don't get it. `hs.Shutdown` should wait until http.Server is really stopped so Sleep shouldn't be required. Also `os.Exit(0)` shouldn't be required as `hs.ListenAndServe()` should return err after `hs.Shutdown` call finishes.

But that's my thoughts, not verified empirically.

Reply:
yep, looks like sleep is not needed if `os.Exit(0)` removed